client/share_trader/js/components/share_trader.js

```diff
@@ -116,10 +40,27 @@ const ShareTrader = React.createClass({
         return states;
     },
 
+    flattenAssetList(assetList) {
+        let flattenedAssetList = [];
+        Object.keys(assetList).forEach((account) => {
+            flattenedAssetList = flattenedAssetList.concat(
+                assetList[account]
+            );
+        });
+        return flattenedAssetList;
+    },
 
     render() {
-        const { activeAccount, accountList, activeAsset, assetList } = this.state;
+        const { activeAccount,
+            accountList,
+            activeAsset,
+            assetList
+        } = this.state;
+
         const states = this.mapAccountsOnStates(accountList);
+        const assetListForAccount =
+            activeAccount && Object.keys(assetList).indexOf(activeAccount.vk) > -1 ?
```

Comment:
Use `assetList.hasOwnProperty(activeAccount.vk)`. The structure / naming of this seems awkward though; if you use `this.flattenAssetList(assetList)`, then that probably isn't the asset list for a single account?